cdap-ui/app/cdap/components/AbstractWidget/JoinTypeWidget/index.tsx

```diff
@@ -0,0 +1,156 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {useState, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import Select from 'components/AbstractWidget/SelectWidget';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import ThemeWrapper from 'components/ThemeWrapper';
+import If from 'components/If';
+
+interface IJoinTypeWidgetProps {
+  value: string;
+  inputSchema: Array<{name: string}>;
+  onChange: (arg0: string) => void;
+}
+
+const DROP_DOWN_OPTIONS: Array<string> = ['Inner', 'Outer'];
+
+export default function JoinTypeWidget({
+  value = '',
+  inputSchema,
+  onChange
+}: IJoinTypeWidgetProps) {
+  const [joinType, setJoinType] = useState(DROP_DOWN_OPTIONS[1]);
+  const [selectedCount, setSelectedCount] = useState(0);
+  const [inputs, setInputs] = useState([]);
+
+  const formatOutput = () => {
+    const outputArr = inputs.filter(schema => schema.selected).map(schema => schema.name);
+    onChange(outputArr.join(','));
+    setSelectedCount(outputArr.length);
+  };
+
+  const joinTypeChange = event => {
+    setJoinType(event.target.value);
+    setInputs(
+      inputSchema.map(input => {
+        return {name: input.name, selected: event.target.value === 'Inner'};
+      })
+    );
+  };
+
+  const checkBoxChange = event => {
+    setInputs(
+      inputs.map(input => {
+        if (input.name === event.target.value) {
+          input.selected = !input.selected;
+        }
+        return input;
+      })
+    );
+  };
+
+  useEffect(() => {
+    const initialModel = value.split(',').map(input => input.trim());
+    if (!value) {
+      setInputs(
+        inputSchema.map(input => {
+          return {name: input.name, selected: false};
+        })
+      );
+    }
+    if (initialModel.length === inputSchema.length) {
+      setJoinType('Inner');
+      setInputs(
+        inputSchema.map(input => {
+          return {name: input.name, selected: true};
+        })
+      );
+    } else {
+      setJoinType('Outer');
+      setInputs(
+        inputSchema.map(input => {
+          return {
+            name: input.name,
+            selected: initialModel.indexOf(input.name) !== -1 ? true : false
+          };
+        })
+      );
+    }
+  }, []);
+
+  useEffect(() => {
+    formatOutput();
+  }, [inputs]);
+
+  return (
+    <ThemeWrapper>
+      {inputs.length > 0 ? (
+        <div className='multi-checkboxes-container'>
```

Comment:
@itsanudeep Can we move these styles now to the css-in-js here in this file? When we remove the less styles this will break right?

Reply:
Sure, will do.